discovery/azure/azure.go

```diff
@@ -419,24 +419,16 @@ func (d *Discovery) refresh(ctx context.Context) (tg *targetgroup.Group, err err
 func (client *azureClient) getVMs(ctx context.Context) ([]virtualMachine, error) {
 	var vms []virtualMachine
 	result, err := client.vm.ListAll(ctx)
-	if err != nil {
```

Comment:
Swallowing the error returned by `ListAll()` doesn't feel very idiomatic IMO. I'd keep this first `if err != nil { ... }` block for readability. And it would also make the for loop more readable as the following error check can happen right after the `NextWithContext()` call.

```go
	for result.NotDone() {
		for _, vm := range result.Values() {
			vms = append(vms, mapFromVM(vm))
		}
		err = result.NextWithContext(ctx)
		if err != nil {
			return nil, fmt.Errorf("could not list virtual machines: %s", err)
		}
	}
```

Reply:
Makes sense. I can do that